tests/proxy/list/list_test.go

```diff
@@ -0,0 +1,388 @@
+// Copyright (c) 2008-2018, Hazelcast, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License")
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package list
+
+import (
+	"github.com/hazelcast/hazelcast-go-client"
+	"github.com/hazelcast/hazelcast-go-client/core"
+	. "github.com/hazelcast/hazelcast-go-client/rc"
+	. "github.com/hazelcast/hazelcast-go-client/tests"
+	"log"
+	"sync"
+	"testing"
+)
+
+var list core.IList
+var client hazelcast.IHazelcastInstance
+var testElement = "testElement"
+
+func TestMain(m *testing.M) {
+	remoteController, err := NewRemoteControllerClient("localhost:9701")
+	if remoteController == nil || err != nil {
+		log.Fatal("create remote controller failed:", err)
+	}
+	cluster, err := remoteController.CreateCluster("3.9", DEFAULT_XML_CONFIG)
+	remoteController.StartMember(cluster.ID)
+	client, _ = hazelcast.NewHazelcastClient()
+	list, _ = client.GetList("myList")
+	m.Run()
+	list.Clear()
+	client.Shutdown()
+	remoteController.ShutdownCluster(cluster.ID)
+}
+
+func TestListProxy_Add(t *testing.T) {
+	defer list.Clear()
+	changed, err := list.Add(testElement)
+	AssertEqualf(t, err, changed, true, "list Add() failed")
+	result, err := list.Get(0)
+	AssertEqualf(t, err, result, testElement, "list Add() failed")
+}
+
+func TestListProxy_AddNilElement(t *testing.T) {
+	defer list.Clear()
+	_, err := list.Add("")
+	AssertNilf(t, err, nil, "list Add() should return error for nil element")
+}
+
+func TestListProxy_AddAt(t *testing.T) {
+	defer list.Clear()
+	list.AddAt(0, testElement)
+	list.AddAt(1, "test2")
+	result, err := list.Get(1)
+	AssertEqualf(t, err, result, "test2", "list AddAt() failed")
+}
+
+func TestListProxy_AddAtNilElement(t *testing.T) {
+	defer list.Clear()
+	err := list.AddAt(0, "")
```

Comment:
this is an empty string, not `nil`.
Alos `AssertNilf(t, err, nil, "..")` checks if third argument is `nil` or not. You already put `nil` there. So you do not test anything here.